adala/runtimes/_openai.py

```diff
@@ -42,16 +43,24 @@ async def async_create_completion(
     Returns:
         Dict[str, Any]: OpenAI response or error message.
     """
-    openai_api_key = openai_api_key or os.getenv("OPENAI_API_KEY")
-    messages = [{"role": "user", "content": user_prompt}]
+    openai_api_key = (
+        openai_api_key or client.api_key or os.environ['OPENAI_API_KEY']
+    )
+    messages = [{'role': 'user', 'content': user_prompt}]
```

Comment:
I know we've talked a bit about this, but is anyone opposed to me just switching adala over to blue?

Reply:
+1